vrx_ign/src/ScoringPlugin.cc

```diff
@@ -0,0 +1,555 @@
+    auto ready_nsec = std::chrono::duration_cast<std::chrono::nanoseconds>(this->readyTime - ready_sec);
+    readyTimeMsg.set_sec(ready_sec.count());
+    readyTimeMsg.set_nsec(ready_nsec.count());
+    this->taskMsgReadyTime.set_allocated_time_value(&readyTimeMsg);
+
+    runningTimeMsg.set_sec(this->runningTime.count());
+
+    auto running_sec = std::chrono::duration_cast<std::chrono::seconds>(this->runningTime);
+    auto running_nsec = std::chrono::duration_cast<std::chrono::nanoseconds>(this->runningTime - running_sec);
+    runningTimeMsg.set_sec(running_sec.count());
+    runningTimeMsg.set_nsec(running_nsec.count());
+    this->taskMsgRunningTime.set_allocated_time_value(&runningTimeMsg);
+    
+    this->taskMsgParam = this->taskMsg.add_param()->mutable_params();
+
+    (*(this->taskMsgParam))["name"] = this->taskMsgName;
+    (*(this->taskMsgParam))["state"] = this->taskMsgState;
+    (*(this->taskMsgParam))["ready_time"] = this->taskMsgReadyTime;
+    (*(this->taskMsgParam))["running_time"] = this->taskMsgRunningTime;
+    (*(this->taskMsgParam))["elapsed_time"] = this->taskMsgElapsedTime;
+    (*(this->taskMsgParam))["remaining_time"] = this->taskMsgRemainingTime;
+    (*(this->taskMsgParam))["timed_out"] = this->taskMsgTimedOut;
+    (*(this->taskMsgParam))["num_collisions"] = this->taskMsgNumCollisions;
+    (*(this->taskMsgParam))["score"] = this->taskMsgScore;
+    
+
+    // this->taskMsg["ready_time"] = builtin_interfaces::msg::Time
+    //                             (rclcpp::Time(this->readyTime.sec,this->readyTime.nsec));
+    // this->taskMsg["running_time"] = builtin_interfaces::msg::Time
+    //                             (rclcpp::Time(this->runningTime.sec,this->runningTime.nsec));
+    this->UpdateTaskMessage();
+
+    // Initialize ROS transport
+    // this->taskPub = rosNode->create_publisher<vrx_ros::msg::Task>(this->taskInfoTopic, 100);
+    // this->contactPub = rosNode->create_publisher<vrx_ros::msg::Contact>(this->contactDebugTopic, 100);
+
+    this->taskPub = std::make_unique<ignition::transport::Node::Publisher>
+        (gzNode->Advertise<ignition::msgs::Param_V>(this->taskInfoTopic));
+    this->contactPub = std::make_unique<ignition::transport::Node::Publisher>
+        (gzNode->Advertise<ignition::msgs::Contact>(this->contactDebugTopic));
+
+    // TODO: Verify - Is this needed in Ignition?
+    // gzNode->Init();
+
+    // TODO: Verify if this is how components of the world are accessed
+    // std::string worldName = _ecm.Component<ignition::gazebo::components::Name>(this->world);
+    std::string worldName = this->world.Name(_ecm).value();
+    
+    std::string collisionTopic =
+        std::string("/gazebosim/") + worldName + std::string("/physics/contacts");
+    if (!gzNode->Subscribe(collisionTopic, &ScoringPlugin::OnCollisionMsg, this))
+    {
+        std::cerr << "Error subscribing to [" << collisionTopic << "]" << std::endl;
+    }
+
```

Comment:
The code in VRX verifies the `VRX_DEBUG` environment variable to set `this->debug`. Should we add it?